contracts/libraries/CircleCCTPAdapter.sol

```diff
@@ -0,0 +1,88 @@
+import "../external/interfaces/CCTPInterfaces.sol";
+
+abstract contract CircleCCTPAdapter {
+    using SafeERC20 for IERC20;
+
+    /**
+     * @notice The domain ID that CCTP will transfer funds to.
+     * @dev This identifier is assigned by Circle and is not related to a chain ID.
+     * @dev Official domain list can be found here: https://developers.circle.com/stablecoins/docs/supported-domains
+     */
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint32 public immutable recipientCircleDomainId;
+
+    /**
+     * @notice The official USDC contract address on this chain.
+     * @dev Posted officially here: https://developers.circle.com/stablecoins/docs/usdc-on-main-networks
+     */
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IERC20 public immutable usdcToken;
+
+    /**
+     * @notice The official Circle CCTP token bridge contract endpoint.
+     * @dev Posted officially here: https://developers.circle.com/stablecoins/docs/evm-smart-contracts
+     */
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    ITokenMessenger public immutable cctpTokenMessenger;
+
+    /**
+     * @notice intiailizes the CircleCCTPAdapter contract.
+     * @param _usdcToken USDC address on the current chain.
+     * @param _cctpTokenMessenger TokenMessenger contract to bridge via CCTP. If the zero address is passed, CCTP bridging will be disabled.
+     * @param _recipientCircleDomainId The domain ID that CCTP will transfer funds to.
+     */
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        IERC20 _usdcToken,
+        ITokenMessenger _cctpTokenMessenger,
+        uint32 _recipientCircleDomainId
+    ) {
+        usdcToken = _usdcToken;
+        cctpTokenMessenger = _cctpTokenMessenger;
+        recipientCircleDomainId = _recipientCircleDomainId;
+    }
+
+    /**
+     * @notice converts address to bytes32 (alignment preserving cast.)
+     * @param addr the address to convert to bytes32
+     * @dev Sourced from the official CCTP repo: https://github.com/walkerq/evm-cctp-contracts/blob/139d8d0ce3b5531d3c7ec284f89d946dfb720016/src/messages/Message.sol#L142C1-L148C6
+     */
+    function _addressToBytes32(address addr) internal pure returns (bytes32) {
+        return bytes32(uint256(uint160(addr)));
+    }
+
+    /**
+     * @notice Returns whether or not the CCTP bridge is enabled.
+     */
+    function _isCCTPEnabled() private view returns (bool) {
+        return address(cctpTokenMessenger) != address(0) && address(usdcToken) != address(0);
+    }
+
+    /**
+     * @notice Transfers USDC from the current domain to the given address on the new domain.
+     * @param to Address to receive USDC on the new domain.
+     * @param amount Amount of USDC to transfer.
+     */
+    function _transferUsdc(address to, uint256 amount) internal {
+        // Require that the CCTP bridge is enabled
+        require(_isCCTPEnabled(), "CCTP bridge is disabled");
```

Comment:
In all of the SpokePools, we first check `_isCCTPEnabled()` before calling into here so the check here is redundant and adds gas. Can we remove this check and add a note to devs that this function will revert if `cctpTokenMessenger` is set incorrectly?